pkg/query-service/rules/apiParams.go

```diff
@@ -44,7 +46,10 @@ func ParsePostableRule(content []byte) (*PostableRule, []error) {
 
 func parsePostableRule(content []byte, kind string) (*PostableRule, []error) {
 	rule := PostableRule{}
+	return parseIntoRule(&rule, content, kind)
+}
 
+func parseIntoRule(rule *PostableRule, content []byte, kind string) (*PostableRule, []error) {
 	var err error
 	if kind == "json" {
 		if err = json.Unmarshal(content, &rule); err != nil {
```

Comment:
Should this just be?
```suggestion
		if err = json.Unmarshal(content, rule); err != nil {
```